### PR TITLE
Add example, fix inferred zone for disk resource policies attachment

### PR DESCRIPTION
**Pull request by emilymye**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/5255

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: Fixed `google_compute_disk_resource_policy_attachment` parsing of region from zone to allow for provider-level zone and make error message more accurate`
```

### CODE REVIEW

File: templates/terraform/pre_delete/compute_disk_resource_policies_attachment.go.erb [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+if err != nil {
+	return err
+}
+
+// resourceProjects are added by zone, read by region, i.e.
 // projects/{project}/regions/{region}/resourcePolicies/{resourceId}
-region := getRegionFromZone(d.Get("zone").(string))
+region := getRegionFromZone(zone)
+if region == "" {
+	return fmt.Errorf("unable to infer region from empty zone")
```
Comment (chrisst) [inline templates/terraform/pre_delete/compute_disk_resource_policies_attachment.go.erb]: Seems a bit strange to assume that zone was empty if region isn't present. What if zone is just a bad string?

Comment (emilymye) [reply templates/terraform/pre_delete/compute_disk_resource_policies_attachment.go.erb]: will fix the message

**Comment by emilymye** [comment]
Overall, I agree that this resource isn't great, but I'll probably try fixing it in a different PR that I could rollback separately. 

I'm not sure region as a computed field helps too much - we'd still need need an encoder/pre-delete no matter what to wrap the data because it's a fine-grained resource for a zonal parent resource (disk) that requires you to set a regional string when you add/delete them, e.g. create/delete is
```
POST zones/{{zone}}/disks/{{disk}}/addResourcePolicies/deleteResourcePolicies {

  "resourcePolicies": {
     "projects/{{project}}/regions/{{region}}/policies/{{name}}": obj,
  }
```
and read is 
```
GET zones/{{zone}}/disks/{{disk}} 
-->

  {
      "name": "adisk"
      "resourcePolicies": [ ... ]
  }
```
which is just GREAT. We'd still need need an encoder/pre-delete no matter what to wrap the data and to set region prior to using it in replaceVars.

**Comment by emilymye** [comment]
deleted my previous comment because it wasn't quite accurate - basically we'd still need to encode (including in pre-delete) the fields into the format compute wants, so I'm not sure having region as a compute field will solve that.

**Pull request merged**